experimental/list_formatter/examples/and_list.rs

```diff
@@ -0,0 +1,23 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use icu_list::{
+    options::{Type, Width},
+    ListFormatter,
+};
+use icu_locid_macros::langid;
+use rand::seq::SliceRandom;
+
+fn main() {
+    let provider = icu_testdata::get_static_provider();
+
+    let list_formatter =
+        ListFormatter::try_new(langid!("es"), &provider, Type::And, Width::Wide).unwrap();
+
+    let mut rng = rand::thread_rng();
```

Comment:
Issue: Please make this deterministic.  These examples are evaluated in Valgrind to count instructions, and it would not be good if two runs produced different instruction counts (e.g., because the regex is evaluated sometimes but not others).

I suggest not using randomization at all and just set a list of strings with a regex-triggering string at the end.  But if you want to use randomization, please use a seed.